lib/inferno/result_summarizer.rb

```diff
@@ -11,6 +11,10 @@ def initialize(results)
     end
 
     def summarize
+      return 'pass' if all_optional_results? &&
+                       unique_result_strings.any?('pass') &&
+                       unique_result_strings.none? { |result| %w[cancel wait running].include? result }
```

Comment:
I don't think cancel belongs here, as the results of the cancelled tests wouldn't prevent the group from passing.

Reply:
removed cancel